File: Tests for EMF and CrossEcore/Eclipse EMF/com.vogella.emf.webpage.model/src-gen/com/vogella/emf/webpage/model/webpage/ArticleType.java
/**
 */
package com.vogella.emf.webpage.model.webpage;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import org.eclipse.emf.common.util.Enumerator;

/**
 * <!-- begin-user-doc -->
 * A representation of the literals of the enumeration '<em><b>Article Type</b></em>',
 * and utility methods for working with them.
 * <!-- end-user-doc -->
 * @see com.vogella.emf.webpage.model.webpage.WebpagePackage#getArticleType()
 * @model
 * @generated
 */
public enum ArticleType implements Enumerator {
	/**
	 * The '<em><b>REVIEW</b></em>' literal object.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @see #REVIEW_VALUE
	 * @generated
	 * @ordered
	 */
	REVIEW(0, "REVIEW", "REVIEW"),

	/**
	 * The '<em><b>STUDY</b></em>' literal object.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @see #STUDY_VALUE
	 * @generated
	 * @ordered
	 */
	STUDY(1, "STUDY", "STUDY");

	/**
	 * The '<em><b>REVIEW</b></em>' literal value.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @see #REVIEW
	 * @model
	 * @generated
	 * @ordered
	 */
	public static final int REVIEW_VALUE = 0;

	/**
	 * The '<em><b>STUDY</b></em>' literal value.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @see #STUDY
	 * @model
	 * @generated
	 * @ordered
	 */
	public static final int STUDY_VALUE = 1;

	/**
	 * An array of all the '<em><b>Article Type</b></em>' enumerators.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @generated
	 */
	private static final ArticleType[] VALUES_ARRAY = new ArticleType[] { REVIEW, STUDY, };

	/**
	 * A public read-only list of all the '<em><b>Article Type</b></em>' enumerators.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @generated
	 */
	public static final List<ArticleType> VALUES = Collections.unmodifiableList(Arrays.asList(VALUES_ARRAY));

	/**
	 * Returns the '<em><b>Article Type</b></em>' literal with the specified literal value.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @param literal the literal.
	 * @return the matching enumerator or <code>null</code>.
	 * @generated
	 */
	public static ArticleType get(String literal) {
		for (int i = 0; i < VALUES_ARRAY.length; ++i) {
			ArticleType result = VALUES_ARRAY[i];
			if (result.toString().equals(literal)) {
				return result;
			}
		}
		return null;
	}

	/**
	 * Returns the '<em><b>Article Type</b></em>' literal with the specified name.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @param name the name.
	 * @return the matching enumerator or <code>null</code>.
	 * @generated
	 */
	public static ArticleType getByName(String name) {
		for (int i = 0; i < VALUES_ARRAY.length; ++i) {
			ArticleType result = VALUES_ARRAY[i];
			if (result.getName().equals(name)) {
				return result;
			}
		}
		return null;
	}

	/**
	 * Returns the '<em><b>Article Type</b></em>' literal with the specified integer value.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @param value the integer value.
	 * @return the matching enumerator or <code>null</code>.
	 * @generated
	 */
	public static ArticleType get(int value) {
		switch (value) {
		case REVIEW_VALUE:
			return REVIEW;
		case STUDY_VALUE:
			return STUDY;
		}
		return null;
	}

	/**
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @generated
	 */
	private final int value;

	/**
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @generated
	 */
	private final String name;

	/**
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @generated
	 */
	private final String literal;

	/**
	 * Only this class can construct instances.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @generated
	 */
	private ArticleType(int value, String name, String literal) {
		this.value = value;
		this.name = name;
		this.literal = literal;
	}

	/**
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @generated
	 */
	@Override
	public int getValue() {
		return value;
	}

	/**
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @generated
	 */
	@Override
	public String getName() {
		return name;
	}

	/**
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @generated
	 */
	@Override
	public String getLiteral() {
		return literal;
	}

	/**
	 * Returns the literal value of the enumerator, which is its string representation.
	 * <!-- begin-user-doc -->
	 * <!-- end-user-doc -->
	 * @generated
	 */
	@Override
	public String toString() {
		return literal;
	}

} //ArticleType

File: Tests for EMF and CrossEcore/CrossEcore/CrossEcore Web Projekt/src/webpage/ArticleType.ts
    export class ArticleType
    {
    	
		public static readonly REVIEW_VALUE:number = 0;
		public static readonly STUDY_VALUE:number = 1;
		
		public static REVIEW:ArticleType = new ArticleType(0, "REVIEW", "REVIEW");
		public static STUDY:ArticleType = new ArticleType(1, "STUDY", "STUDY");

		private static VALUES_ARRAY:Array<ArticleType> = [
			ArticleType.REVIEW, 
			ArticleType.STUDY
		];

        public static get_string(literal:string):ArticleType
        {
            for (let i = 0; i < this.VALUES_ARRAY.length; i++)
            {
                let result = this.VALUES_ARRAY[i];
                if (result.toString() === literal)
                {
                    return result;
                }
            }
            return null;
        }

        public static getByName(name:string):ArticleType
        {
	        for (let i = 0; i < this.VALUES_ARRAY.length; i++)
	        {
	            let result = this.VALUES_ARRAY[i];
	            if (result.getName()==name)
	            {
	                return result;
	            }
	        }
	        return null;
        }

        public static get_number(value:number):ArticleType
        {
            switch (value)
            {
			case this.REVIEW_VALUE: return this.REVIEW;
			case this.STUDY_VALUE: return this.STUDY;
            }
            return null;
        }

	    private value:number;
	    private name:string;
	    private literal:string;

	    private constructor(value:number, name:string, literal:string)
	    {
	        this.value = value;
	        this.name = name;
	        this.literal = literal;
	    }
	
	    public getLiteral():string
	    {
	        return this.literal;
	    }
	
	    public getName():string
	    {
	        return this.name;
	    }
	
	    public getValue():number
	    {
	        return this.value;
	    }
	    
	    public toString():string
	    {
	        return this.literal;
	    }
    }
